src/dodal/devices/motors.py

```diff
@@ -2,60 +2,147 @@
 from ophyd_async.epics.motor import Motor
 
 
-class XYZPositioner(StandardReadable):
+class XThetaStage(StandardReadable):
     """
 
-    Standard ophyd_async xyz motor stage, by combining 3 Motors,
-    with added infix for extra flexibliy to allow different axes other than x,y,z.
+    ophyd_async x theta motor stage, combining 2 Motors: a horizontal stage and a
+    rotational stage mutually perpendicular to each other and the beam.
 
     Parameters
     ----------
     prefix:
-        EPICS PV (Common part up to and including :).
+        Common part of the EPICS PV for all motors, including ":".
     name:
-        name for the stage.
-    infix:
-        EPICS PV, default is the ("X", "Y", "Z").
-    Notes
-    -----
-    Example usage::
-        async with init_devices():
-            xyz_stage = XYZPositioner("BLXX-MO-STAGE-XX:")
-    Or::
-        with init_devices():
-            xyz_stage = XYZPositioner("BLXX-MO-STAGE-XX:", infix = ("A", "B", "C"))
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", x_infix: str = "X", theta_infix: str = "A"
+    ):
+        with self.add_children_as_readables():
+            self.x = Motor(prefix + x_infix)
+            self.theta = Motor(prefix + theta_infix)
+        super().__init__(name=name)
+
+
+class XYStage(StandardReadable):
+    """
+
+    ophyd_async x y motor stage, combining 2 Motors, mutually perpendicular to the beam.
+
+    Parameters
+    ----------
+    prefix:
+        Common part of the EPICS PV for all motors, including ":".
+    name:
+        Name of the stage, each child motor will be named "{name}-{field_name}"
+    *_infix:
+        Infix between the common prefix and the EPICS motor record fields for the field.
+
+    """
+
+    def __init__(
+        self, prefix: str, name: str = "", x_infix: str = "X", y_infix: str = "Y"
+    ):
+        with self.add_children_as_readables():
+            self.x = Motor(prefix + x_infix)
+            self.y = Motor(prefix + y_infix)
+        super().__init__(name=name)
```

Comment:
Or should these be unified as TwoAxisStage, with self.fast_axis and self.slow_axis and configuration to decide what it is called on read? I just would like to avoid the construction of device classes where you do not know what attrs each instance has, to aid in plan development.